docs/content/docs/dev/table/data_stream_api.md

```diff
@@ -337,6 +324,86 @@ In particular, the section discusses how to influence the schema derivation with
 types. It covers working with event-time and watermarks. It discusses how to declare a primary key and
 changelog mode for the input and output streams.
 
+The example above shows how the final result is computed incrementally by continuously emitting row-wise
+updates for each incoming record. However, in cases where the input streams are finite (i.e. *bounded*),
+a result can be computed more efficiently by leveraging batch processing principles. Both DataStream
```

Comment:
Perhaps a brief explanation of what this means or a link to more info?

Reply:
I added a concrete example and a link to the dynamic table section.